granary/reddit.py

```diff
@@ -45,7 +54,10 @@ def praw_to_actor(self, praw_user):
     Returns:
       an ActivityStreams actor dict, ready to be JSON-encoded
     """
-    user = reddit.praw_to_user(praw_user)
+    try:
+      user = reddit.praw_to_user(praw_user)
```

Comment:
hmm, `praw_to_user` doesn't raise anything, right?

https://github.com/snarfed/oauth-dropins/blob/b72522b94bae9dcef08901717d78fcca78e19736/oauth_dropins/reddit.py#L130-L137

i expect the `NotFound`s are coming from the `r.*()` calls in `get_activities_response()` and `fetch_replies()` instead. example log: https://brid.gy/log?start_time=1588205455&key=agdicmlkLWd5chULEgZSZWRkaXQiCXNjaG5hcmZlZAw

maybe filling in comment authors? near the end i see https://oauth.reddit.com/comments/1q90uz/ , which 200s, and a few later, https://oauth.reddit.com/user/crossmyheartxx/about/ 404s. so maybe this:

https://github.com/snarfed/granary/blob/48afa321c30e40149f1933f98eff4590f5899d6b/granary/reddit.py#L193

i still don't understand why this happens, though.

Reply:
yeah, from that example, if you follow the stack trace down it goes to 
```File "/srv/oauth_dropins/reddit.py", line 133, in praw_to_user
    'subreddit': user.subreddit,```

Reply:
thanks! and sorry for not following more closely before, you'd already followed the stack trace better than me the first time.

Reply:
looks like it's from the `subreddit` attribute specifically. maybe we narrow the `try/except` to just that? and omit it if it's not there?

actually, more importantly, why do we want a subreddit for a user object at all? aren't users independent of subreddits? i see that we use it in `user_to_actor()`, for both user url and web site links, but why?

Reply:
haha, yeah. it's kind of strange but apparently (based on https://praw.readthedocs.io/en/latest/code_overview/models/redditor.html) the way reddit stores a user profile description is by giving each person their own "subreddit."  I'll add a note about that above it because I agree it seems highly circuitous. 

As for adding the try/except there, that probably will work better.  I'll make that change.

Reply:
when testing a little more, it appears that all of the attribute calls in `praw_to_user` throw 404s whenever the user's account has been deleted.  So it seems we need to try/except the whole function and replace with a missing user (or do something else).

I'll still add some more documentation to explain why in these functions.